smt/available_solvers.cpp

```diff
@@ -174,6 +178,10 @@ SmtSolver create_solver_for(
   assert(s);
   if (ic3_engine) {
     s->set_opt("produce-unsat-assumptions", "true");
+    if (se == BTOR && witness) {
+      // Witness generation in IC3 needs reset_assertions support.
+      s->set_opt("base-context-1", "true");
+    }
```

Comment:
Not only witness generation but also some engines rely on `reset_assertions`. I think in Pono we should simply enable this by default for Boolector.

While this may impact performance, it avoids adding more logic for special cases and makes the code easier to maintain. Since Boolector is no longer our primary solver, I think this tradeoff is acceptable.

Reply:
Good point.